sql/src/test/java/io/crate/concurrent/ContextCompletionListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.concurrent;
+
+import com.google.common.util.concurrent.SettableFuture;
+import io.crate.concurrent.CompletionListener;
+import io.crate.concurrent.CompletionState;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class ContextCompletionListener implements CompletionListener {
+
+    private SettableFuture<Void> future = SettableFuture.create();
+
+    public SettableFuture<Void> future() {
```

Comment:
since this seems to be just for testing, would it make sense to just expose a `.get()` that always calls the future with a timeout? That way it would be ensured that tests can't get stuck